### PR TITLE
Move Build System to .NET Core 2.1, projects to .NET Standard 2.0 and .NET 4.5.2

### PR DESCRIPTION
**Pull request by Aaronontheweb**
This PR introduces the following changes:

1. Standardize most .NET Framework / Standard / Core versions used inside the core / contrib libraries, their unit tests, MNTR, and so forth. These versions are now set inside `common.props` and referenced via build parameters.
2. Remove `dotnet-xunit`, given that it's deprecated and won't be supported any further in the future. 
3. Replace `dotnet-xunit` with `dotnet test` inside our build system, which has future-facing support for the foreseeable future. All of our current build parameters (such as `-parallel none`) that we rely on inside XUnit are still fully supported.
4. Fix the long-standing issues with VSTest and Visual Studio locking up the solution whenever certain test suites are run - turns out this is caused by .NET 4.5.2 and possibly older versions of .NET and some of the changes introduced into more recent versions of the VSTest adapter system. Upgrading our unit tests to target .NET 4.6.1 at a minimum fixes the issue.
5. Update some of our plugins that were supporting really random versions of .NET, like .NET 4.5.1. 

All testkits now target .NET 4.5.2 - consistent with what XUnit supports.
All libraries still target .NET 4.5 and .NET Standard 1.6, same as before.
All unit tests, MNTR, perf tests, and so forth target .NET 4.6.1 and .NET Core 1.1.

I have another set of changes planned to get us running on the .NET Core 2.1.500 SDK, but I want to restrict the number of changes made at anyone time and this PR already touches a lot of different projects.

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Based on build system output thus far:
* Windows build - ok
* MNTR .NET Core - ok
* MNTR .NET full - ok
* Unit Tests .NET Core / Windows - ok but muted test failures cause process to abort; need to add back the smarter try...catch error handling we had in FAKE
* Unit Tests .NET Full / Windows - **hung** on Akka.Cluster.Tests after 30 minutes. Same issue with VSTest adapters in Windows. Upgrading from .NET 4.5.2 was supposed to address this. Might also have to address the .NET Core SDK and XUnit versions to fix this.
* Unit Tests .NET Core / Linux - **compilation failure** due to missing F# dependencies on Akka.FSharp.
* Performance tests Windows / .NET full - ok but still racy (NBench issue)
* NuGet pack - **fails** on MNTR (still looking .NET 4.5.2 directory)

**Comment by Aaronontheweb** [comment]
Going to work the problem starting with the hung .NET Framework tests, since that's been the single biggest pain in the ass for everyone who's been complaining about ReSharper issues inside Visual Studio. I need to do some analysis and see which projects it is that hangs just in-case I can't fix it by upgrading the build + test runtime we're using.

Essentially, in the span of a year all of the .NET Core 1.1 infrastructure has byte-rotted immensely and this is what we're now paying for.

**Comment by Aaronontheweb** [comment]
https://github.com/Microsoft/msbuild/issues/2275

**Comment by Aaronontheweb** [comment]
Ok, yeah - can confirm the hang in the VSTest adapter was caused by FsCheck. Unfortunately, upgrading to the latest (2.13.0) didn't solve the issue. But removing the FSCHECK compilation symbol from .NET 4.6.1 builds did. 

Going to do that and have us skip FsCheck specs for the time being until we can get that regression fixed.

**Comment by Aaronontheweb** [comment]
I think I got all of the FsCheck instances in this case. Just need to add back the `FSCHECK` compilation symbol for .NET 4.6.1 builds to restore this functionality again.

**Comment by Aaronontheweb** [comment]
Akka.Persistence specs OOM-ing on .NET 4.6.1

**Comment by Aaronontheweb** [comment]
Wonder if this is caused by the `EventMemoryJournal` filling up and never leaving memory

**Comment by Aaronontheweb** [comment]
OOM caused

```
Failed   Akka.Persistence.Tests.AtLeastOnceDeliveryReceiveActorSpec.PersistentReceive_must_deliver_messages_in_order_whe
n_nothing_is_lost                                                                                                       
Error Message:                                                                                                          
 System.OutOfMemoryException : Exception of type 'System.OutOfMemoryException' was thrown.                              
Stack Trace:                                                                                                            
   at System.String.Concat(String str0, String str1, String str2)                                                       
   at Akka.Configuration.Hocon.Parser.ParseObject(HoconValue owner, Boolean root, String currentPath)                   
   at Akka.Configuration.Hocon.Parser.ParseKeyContent(HoconValue value, String currentPath)                             
   at Akka.Configuration.Hocon.Parser.ParseObject(HoconValue owner, Boolean root, String currentPath)                   
   at Akka.Configuration.Hocon.Parser.ParseKeyContent(HoconValue value, String currentPath)                             
   at Akka.Configuration.Hocon.Parser.ParseObject(HoconValue owner, Boolean root, String currentPath)                   
   at Akka.Configuration.Hocon.Parser.ParseKeyContent(HoconValue value, String currentPath)                             
   at Akka.Configuration.Hocon.Parser.ParseObject(HoconValue owner, Boolean root, String currentPath)                   
   at Akka.Configuration.Hocon.Parser.ParseKeyContent(HoconValue value, String currentPath)                             
   at Akka.Configuration.Hocon.Parser.ParseObject(HoconValue owner, Boolean root, String currentPath)                   
   at Akka.Configuration.Hocon.Parser.ParseText(String text, Func`2 includeCallback)                                    
   at Akka.Configuration.ConfigurationFactory.FromResource(String resourceName, Assembly assembly)                      
   at Akka.Configuration.ConfigurationFactory.FromResource(String resourceName)                                         
   at Akka.Actor.Settings..ctor(ActorSystem system, Config config)                                                      
   at Akka.Actor.Internal.ActorSystemImpl..ctor(String name, Config config)                                             
   at Akka.TestKit.TestKitBase.InitializeTest(ActorSystem system, Config config, String actorSystemName, String testActo
rName)                                                                                                                  
   at Akka.TestKit.TestKitBase..ctor(ITestKitAssertions assertions, ActorSystem system, Config config, String actorSyste
mName, String testActorName)                                                                                            
   at Akka.TestKit.AkkaSpec..ctor(Config config, ITestOutputHelper output)                                              
   at Akka.Persistence.Tests.PersistenceSpec..ctor(Config config, ITestOutputHelper output)                             
   at Akka.Persistence.Tests.AtLeastOnceDeliveryReceiveActorSpec..ctor()                                                
```

**Comment by Aaronontheweb** [comment]
I suspect this issue is related to this: https://github.com/akkadotnet/akka.net/issues/2640

Looks like it's an issue with the `ForkJoinDispatcher`.

**Comment by Aaronontheweb** [comment]
Since #3734 has been merged in, re-testing this to see if the OOM exception in the `Akka.Persistence.Tests` library still persists. Tried it locally on my machine and it appears as though the issue was resolved.

**Comment by Aaronontheweb** [comment]
Looks like an FSharp-related compilation failure on Linux

**Comment by Aaronontheweb** [comment]
Looks like also a config issue with the way the `Akka.Streams` testkit specs are setup. Looking into that next.

**Comment by Aaronontheweb** [comment]
Upgraded to the 2.1 NET Core SDK after all - resolves issue with `fsproj`s not being compiled when built from the `.sln` file on Linux.

**Comment by Aaronontheweb** [comment]
Can't bump to 2.1 SDK without dealing with the usual .NET Standard 2.0 issues -- going to see if I can just hack the FSharp issues for now.

**Comment by Aaronontheweb** [comment]
If this doesn't work, we'll simply drop the `.fsproj` from solution file until 1.3.12 is finished, then immediately add them back once we start the 1.4 train as soon as the 1.3.12 bits are out the door. Can't hold up progress forever working around bugs on old tools.

**Comment by Aaronontheweb** [comment]
Build system now works but there are some flakey specifications at work again. Going to double check to make sure that there's nothing systematically wrong (i.e. the build system is broken) but I feel pretty good about the results thus far.

**Comment by Aaronontheweb** [comment]
Looks like we have some Akka.Streams.Dsl specs failing consistently on .NET Core - going to take a look at those.

**Comment by Aaronontheweb** [comment]
Was able to verify that the issue with the Streams specs hitting this error:

https://github.com/akkadotnet/akka.net/blob/41c116d2a3d280481e584401608ec848dc147632/src/core/Akka.Streams.TestKit.Tests/ScriptedTest.cs#L249

"Too many idle rounds"

Is not caused by changing Akka.Streams from .NET Standard 1.6 to 2.0. If anything, it appears to be an issue solely related to changing the test suite from .NET Core 1.1 to .NET Core 2.1. The issue doesn't occur on .NET Framework 4.6.1 at all, but it does appear on both Linux and Windows on .NET Core 2.1 but not 1.1.

I suspect it's a timing issue related to performance changes in the underlying platform; .NET Core 2.1 is significantly faster than .NET 4.6.1 and .NET Core 1.1.

I can't point to anything specific but that's the best explanation so far.

**Comment by Aaronontheweb** [comment]
```
Akka.Tests.Util.ByteStringSpec.A_ByteString_must_be_sequential_when_slicing_from_index] System.TypeInitializationException : The type initializer for '<StartupCode$FsCheck>.$Arbitrary' threw an exception.
---- System.IO.FileLoadException : Could not load file or assembly 'FSharp.Core, Version=4.4.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)
-------- System.IO.FileLoadException : Could not load file or assembly 'FSharp.Core, Version=4.4.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)
```

Fun with strong naming between FsCheck and FSharp.Core...

**Comment by Aaronontheweb** [comment]
```
Unable to load DLL 'sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
[19:36:34] :		 [RunTests] Cause: [akka://SqliteSnapshotStoreSpec/system/akka.persistence.snapshot-store.sqlite#893941320]: Akka.Actor.ActorInitializationException: Exception during creation ---> System.DllNotFoundException: Unable to load DLL 'sqlite3': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
```

Looks like DLL issues with the SQLite drivers

**Comment by Aaronontheweb** [comment]
Things are looking MUCH better in the .NET Framework test suite now. Had to drop Akka.Persistence.Sqlite specs for the time being due to a native DLL loading exception that doesn't occur on .NET Core - going to test some of our SQLite samples though and see if they still work prior to signing off on that work.

File: src/core/Akka.Streams.TestKit.Tests/ScriptedTest.cs [hunk 1]
```diff
@@ -200,7 +200,7 @@ public void Request(int demand)
 
             public bool ShakeIt()
             {
-                var oneMilli = TimeSpan.FromMilliseconds(1);
+                var oneMilli = TimeSpan.FromMilliseconds(10);
```
Comment (Aaronontheweb) [inline src/core/Akka.Streams.TestKit.Tests/ScriptedTest.cs]: So the issue, near as I can tell based on the additional debug logging I added: the `ReceiveWhile` never even had a chance to run. Bails out immediately in some small cases. I can't figure out what caused this to happen between .NET Core 1.1 and 2.1 but I'm not going to wait forever to get to the bottom of it.

**Comment by Aaronontheweb** [comment]
Not quite ready to merge this in yet - going to try to push out a v1.3.13 with some bug fixes first. Once this gets merged in, we're on the v1.4.0 train for good.

**Comment by Aaronontheweb** [comment]
Pulling in the latest Akka.NET v1.3.13 bits before we merge this into the `dev` branch and "burn the boats" on any more 1.3.* changes.

**Comment by heatonmatthew** [comment]
@Aaronontheweb I just pushed a commit that removes some of the deficiencies of the 1.x netstandard api which are now available in the 2.0 netstandard api. Serialization is a big one.

I'll take a look into it, but this might help with getting #3725 running on DotNet Core.

**Comment by heatonmatthew** [comment]
@Aaronontheweb I cherry-picked your changes from #3725 on top of my two commits above and both the Framework and Core tests appear to work on windows. I'll spin it up in a Linux docker container tomorrow at work and given them a run there too, but I think having the serialization attribute be the platform one (rather than the Akka polyfil) has helped. 

This is incidentally one of the core factors that caused me to start down this netstandard2.0 path. I was doing testing with the 3.0 preview 5 and things were getting a bit confused with the significant differences to the 1.x standard api. :)

**Comment by Aaronontheweb** [comment]
@heatonmatthew great! I need to work on that ASAP as that bug is having a live impact on some of our customers - would you mind sending in a pull request so I can see the changes?

**Comment by Aaronontheweb** [comment]
@heatonmatthew as for those commits you want to put on top of this PR - would love to see a PR for that too. Would be very much appreciated!

**Comment by heatonmatthew** [comment]
@Aaronontheweb PRs created: 

- #3790 removes the netstandard1.x restrictions
- #3791 cherry-picks your Akka.Cluster and Akka.Remote changes on top of them

**Comment by Aaronontheweb** [comment]
@heatonmatthew thanks Matt! reviewed #3790 - I'll take a look at #3791 soon too. Really appreciate your help.

**Pull request merged**